client/blocks/login/two-factor-authentication/verification-code-input.jsx

```diff
@@ -0,0 +1,149 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import ExternalLink from 'components/external-link';
+import FormButtonsBar from 'components/forms/form-buttons-bar';
+import FormButton from 'components/forms/form-button';
+import FormTextInput from 'components/forms/form-text-input';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormCheckbox from 'components/forms/form-checkbox';
+import FormLabel from 'components/forms/form-label';
+import FormInputValidation from 'components/forms/form-input-validation';
+import Card from 'components/card';
+import { localize } from 'i18n-calypso';
+import { loginUserWithTwoFactorVerificationCode, internalRedirect } from 'state/login/actions';
+import { getVerificationCodeSubmissionError } from 'state/login/selectors';
+import {
+	getTwoFactorAuthId,
+	getTwoFactorAuthNonce,
+	isLoginSuccessful
+} from 'state/login/selectors';
+
+class VerificationCodeInput extends Component {
+	state = {
+		twostep_code: '',
+		remember: false,
+		error: null
+	};
+
+	componentWillMount() {
+		this.setState( {
+			error: this.props.error
+		} );
+	}
+
+	componentWillReceiveProps( newProps ) {
+		this.setState( {
+			error: newProps.error
+		} );
+	}
+
+	componentDidUpdate() {
+		if ( this.props.isLoginSuccessful ) {
+			window.location.href = ( this.props.redirectLocation || '/' );
+		}
+	}
+
+	onChangeField = ( event ) => {
+		if ( event.target.type === 'checkbox' ) {
+			this.setState( {
+				[ event.target.name ]: event.target.checked
+			} );
+			return;
+		}
+		// Reset the error state if the user updates the field after an error coming
+		// from the state
+		this.setState( {
+			[ event.target.name ]: event.target.value,
+			error: null
+		} );
+	};
+
+	onCodeSubmit = ( e ) => {
+		const { twostep_id, twostep_nonce } = this.props;
+		const { twostep_code, remember } = this.state;
+		e.preventDefault();
+		this.props.loginUserWithTwoFactorVerificationCode( twostep_id, twostep_code, twostep_nonce, remember );
+	}
+
+	render() {
+		const isError = !! this.state.error;
+		let errorText = this.state.error;
+		if ( isError ) {
+			errorText = this.props.translate( 'Invalid verification code' );
+		}
+		return (
+			<div>
+				<form onSubmit={ this.onCodeSubmit }>
+					<Card>
+						<p>
+							{ this.props.translate( 'Please enter the verification code generated' +
```

Comment:
Hi! I've found a possible matching string that has already been translated **27** times:
`translate( 'Please enter the verification code generated by your Authenticator mobile application.' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).